### PR TITLE
Create `evms.info` static table

### PR DESCRIPTION
**Pull request by hildobby**
This PR creates a static `evms.info` table that contains high level information on each EVM/L2 available on Dune. I tend to hardcoded those a lot into my queries so this probably much easier and I'm sure others will also find it useful.

# SPELLBOOK FREEZE

From June 22 to June 29, we will be freezing some Spellbook contributions for the migration to DuneSQL. We will not accept contributions to the lineage of the following models:

* dex.trades
* nft.trades
* labels
* token.erc20
* tokens.nft

Run the following command to see the list of affected files:

```
dbt ls --resource-type model --output path --select +dex_trades +labels +nft_trades +tokens_nft +tokens_erc20
```

Don't hesitate to reach out on Discord if you have any questions.

Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/evms/evms_info.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+                                    "sector",
+                                    "evms",
+                                    \'["hildobby"]\') }}')
+}}
+
+SELECT chain_id, blockchain, name, chain_type, rollup_type, native_token_symbol, wrapped_native_token_address, explorer_link, first_block_time
+FROM (VALUES
+        (1, 'ethereum', 'Ethereum', 'Layer 1', NULL, 'ETH', 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2, 'https://etherscan.io', timestamp '2015-07-30 15:26')
+        , (43114, 'avalanche_c', 'Avalanche', 'Layer 1', NULL, 'AVAX', 0xb31f66aa3c1e785363f0875a1b74e27b85fd66c7, 'https://optimistic.etherscan.io', timestamp '2020-09-23 11:02')
+        , (42161, 'arbitrum', 'Arbitrum', 'Layer 2', 'Optimistic Rollup', 'ETH', 0x82af49447d8a07e3bd95bd0d56f35241523fbab1, 'https://arbiscan.io', timestamp '2021-05-29 00:35')
```
Comment (MSilb7) [inline models/evms/evms_info.sql:14]: Arbitrum One

File: models/evms/evms_info.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        , (10, 'optimism', 'Optimism', 'Layer 2', 'Optimistic Rollup', 'ETH', 0x4200000000000000000000000000000000000006, 'https://optimistic.etherscan.io', timestamp '2021-11-11 21:16')
```
Comment (MSilb7) [inline models/evms/evms_info.sql:15]: OP Mainnet

**Comment by MSilb7** [comment]
Left some comments on chain names. IMO we should be as explicit as possible (i.e. testnet vs mainnet, chain release) since these top-level names will have many chains underneath them (i.e. many Arbitrum chains, many Optimism chains, many Polygon chains).

There is a similar table here: https://github.com/duneanalytics/spellbook/blob/main/models/chain_info/chain_info_dune_chains.sql

Which has a PR in to update it (adding similar info & Full Chain Names): https://github.com/duneanalytics/spellbook/pull/3607/files#diff-0c24666de5dca48068c38e34996a489b6162d15bee068e7df21f81e462973ff0

So I think we need to rally around some schema / table name for this kind of thing (agree we need it).

File: models/evms/evms_info.sql [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        post_hook='{{ expose_spells(\'["ethereum", "polygon", "bnb", "avalanche_c", "gnosis", "fantom", "optimism", "arbitrum", "celo"]\',
+                                    "sector",
+                                    "evms",
+                                    \'["hildobby"]\') }}')
+}}
+
+SELECT chain_id, blockchain, name, chain_type, rollup_type, native_token_symbol, wrapped_native_token_address, explorer_link, first_block_time
+FROM (VALUES
+        (1, 'ethereum', 'Ethereum', 'Layer 1', NULL, 'ETH', 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2, 'https://etherscan.io', timestamp '2015-07-30 15:26')
+        , (43114, 'avalanche_c', 'Avalanche', 'Layer 1', NULL, 'AVAX', 0xb31f66aa3c1e785363f0875a1b74e27b85fd66c7, 'https://optimistic.etherscan.io', timestamp '2020-09-23 11:02')
```
[the remark this answers is not in the document]
Comment (MSilb7) [reply models/evms/evms_info.sql:13]: & block explorer should be snowtrace

File: models/evms/evms_info.sql (hunk 2 above)
Comment (MSilb7) [reply models/evms/evms_info.sql:15]: Another comment - the legacy OVM1 started July 2021, but the current system is Nov 11 2021.

Maybe we add a row for optimism_ovm1_legacy ?

**Comment by hildobby** [comment]
> Left some comments on chain names. IMO we should be as explicit as possible (i.e. testnet vs mainnet, chain release) since these top-level names will have many chains underneath them (i.e. many Arbitrum chains, many Optimism chains, many Polygon chains).
> 
> I think we should also [add Goerli ](https://github.com/duneanalytics/spellbook/pull/3607) & not separate out mainnets from testnets (having to union = bad experience).
> 
> There is a similar table here: https://github.com/duneanalytics/spellbook/blob/main/models/chain_info/chain_info_dune_chains.sql
> 
> Which has a PR in to update it (adding similar info & Full Chain Names): https://github.com/duneanalytics/spellbook/pull/3607/files#diff-0c24666de5dca48068c38e34996a489b6162d15bee068e7df21f81e462973ff0
> 
> So I think we need to rally around some schema / table name for this kind of thing (agree we need it).

I agree on the more explicit names, just addressed those.

I think the `evms` sector is more appropriate than the `chain_info` one for this given that this applies specifically to EVM chains and its L2s. Also this makes it easier to access when using the evms sector table (which is how I plan to use it mostly).

Imo the `chain_info` (that I would just rename to a simple `chains`) sector should feed from `evms.info` and also contain info from Solana & Bitcoin (all chains on Dune, regardless if EVM based or not).

I still think adding testnets to the evms table might be confusing to a lot of wizards (who may interpret those as mainnet data) and thats why I'm waiting for at least 1 more testnet to have its data on dune before making `evms_testnets`. A good compromise could be to have a `evms_and_testnets` sector that is all views of `evms` UNION ALL `evms_testnets`. What do you think?

**Comment by MSilb7** [comment]
> > Left some comments on chain names. IMO we should be as explicit as possible (i.e. testnet vs mainnet, chain release) since these top-level names will have many chains underneath them (i.e. many Arbitrum chains, many Optimism chains, many Polygon chains).
> > I think we should also [add Goerli ](https://github.com/duneanalytics/spellbook/pull/3607) & not separate out mainnets from testnets (having to union = bad experience).
> > There is a similar table here: https://github.com/duneanalytics/spellbook/blob/main/models/chain_info/chain_info_dune_chains.sql
> > Which has a PR in to update it (adding similar info & Full Chain Names): https://github.com/duneanalytics/spellbook/pull/3607/files#diff-0c24666de5dca48068c38e34996a489b6162d15bee068e7df21f81e462973ff0
> > So I think we need to rally around some schema / table name for this kind of thing (agree we need it).
> 
> I agree on the more explicit names, just addressed those.
> 
> I think the `evms` sector is more appropriate than the `chain_info` one for this given that this applies specifically to EVM chains and its L2s. Also this makes it easier to access when using the evms sector table (which is how I plan to use it mostly).
> 
> Imo the `chain_info` (that I would just rename to a simple `chains`) sector should feed from `evms.info` and also contain info from Solana & Bitcoin (all chains on Dune, regardless if EVM based or not).
> 
> I still think adding testnets to the evms table might be confusing to a lot of wizards (who may interpret those as mainnet data) and thats why I'm waiting for at least 1 more testnet to have its data on dune before making `evms_testnets`. A good compromise could be to have a `evms_and_testnets` sector that is all views of `evms` UNION ALL `evms_testnets`. What do you think?

Yeah I think either: 1) Have chain_info read from evms (to your point) or 2) deprecate the static chain_info table (but keep the export from chainlist).

A segmented testnet schema that then unifies in to a mainnets + testnets schema makes sense. `evms_and_testnets` feels clunky as a name. I imagine only a super small segment (i.e. me and you) are using evms now, so we could rename without worrying about not breaking existing dashboards.

I'll have spells using mainnets & testnets (L1 goerli for now, hoping we'll have more) coming, so good to figure this out sooner rather than later.

**Comment by hildobby** [comment]
> Yeah I think either: 1) Have chain_info read from evms (to your point) or 2) deprecate the static chain_info table (but keep the export from chainlist).

Yeah i agree here

> A segmented testnet schema that then unifies in to a mainnets + testnets schema makes sense. `evms_and_testnets` feels clunky as a name. I imagine only a super small segment (i.e. me and you) are using evms now, so we could rename without worrying about not breaking existing dashboards.

Not sure how small the segment of users actually is, got quite a few comments/DMs following this PR:
https://twitter.com/hildobby_/status/1668553182568390657?s=20
Andrew will look into `evms` sector usage when he is back from his break and we can decide with that info

> I'll have spells using mainnets & testnets (L1 goerli for now, hoping we'll have more) coming, so good to figure this out sooner rather than later.

Yupp, that's the next step after this PR and [this one](https://github.com/duneanalytics/spellbook/pull/3640).

**Review by alokeutpal — APPROVED** [review]

**Review by Simeon221 — APPROVED** [review]
1ea2c2f84a3b1216ed27e69cef64c67e7a5eff45

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**